src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/__tests__/ChartBoundaryLevelsConfiguration.test.tsx

```diff
@@ -119,23 +120,28 @@ describe('ChartBoundaryLevelsConfiguration', () => {
     );
 
     expect(screen.getByLabelText('Default boundary level')).not.toHaveValue();
-    const boundaryLevels = within(
+    const defaultBoundaryLevelOptions = within(
       screen.getByLabelText('Default boundary level'),
     ).getAllByRole('option');
 
-    expect(boundaryLevels).toHaveLength(4);
-    expect(boundaryLevels[0]).toHaveTextContent('Please select');
-    expect(boundaryLevels[0]).toHaveValue('');
-    expect(boundaryLevels[1]).toHaveTextContent('Boundary level 1');
-    expect(boundaryLevels[1]).toHaveValue('1');
-    expect(boundaryLevels[2]).toHaveTextContent('Boundary level 2');
-    expect(boundaryLevels[2]).toHaveValue('2');
-    expect(boundaryLevels[3]).toHaveTextContent('Boundary level 3');
-    expect(boundaryLevels[3]).toHaveValue('3');
+    expect(defaultBoundaryLevelOptions).toHaveLength(
+      testMeta.boundaryLevels.length + 1,
+    );
+    defaultBoundaryLevelOptions.forEach((option, index) => {
+      if (index === 0) {
+        expect(option).toHaveTextContent('Please select');
+        expect(option).toHaveValue('');
+      } else {
+        const { id, label } = testMeta.boundaryLevels[index - 1];
+        expect(option).toHaveTextContent(label);
+        expect(option).toHaveValue(String(id));
+      }
+    });
```

Comment:
Wouldn't advise using for loops with conditional logic like this in test cases. We want things to be as clear as possible even at the expense of a bit of copy and paste.

Keep things simple and just check things using indices.

Reply:
as above 
I've simplified the code to be more obvious, stating that the options should match up with the supplied boundary levels in the data - I've removed the conditional in the loop. I'll take this forward, keeping tests simple and obvious, avoiding loops with the idea saving lines isn't worth the complexity in tests.